functions/zoom-downloader.py

```diff
@@ -129,11 +130,12 @@ class Download:
 
     def __init__(self, sqs, data):
         self.sqs = sqs
-        self.data = data
+        self.download_message = deepcopy(data)  # do not modify
```

Comment:
We need a copy of the original message to send to the dlq. Dlq messages were failing because self.data was being modified to include a datetime object that was not JSON serializable ("created_local") but I don't want to make that into a string because we actually do want to use the datetime object in a few places and not be constantly converting back and forth.

Reply:
this reminds me of something i feel like i've lost sight of: we are creating the dead letter queues separate from the functions and then manually putting messages in them in the case of failures, but the lambda creation supports a DLQ as a [construct property](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-lambda.Function.html#deadletterqueue) and will write events to it automatically (after retries expire). See https://aws.amazon.com/about-aws/whats-new/2016/12/aws-lambda-supports-dead-letter-queues/. So what is going on here? Was our implementation a workaround for something?

Reply:
It was for sending things to deadletter that we knew were not retryable... maybe it's not worth it though